workspace/src/main/kotlin/com/cosmotech/workspace/service/WorkspaceServiceImpl.kt

```diff
@@ -543,6 +546,26 @@ internal class WorkspaceServiceImpl(
     this.eventPublisher.publishEvent(
         DeleteHistoricalDataWorkspace(this, organizationId, it.id!!, data.deleteUnknown))
   }
+
+  fun checkReadSecurity(workspace: Workspace): Workspace {
```

Comment:
Same remark, could you check the `CsmRbac.check(rbacSecurity,READ_SECURITY_PERMISSION)` instead